### ISSUE TITLE
`null` initializer for non-nullable property (or field) should warn

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#26628]
Similar to https://github.com/dotnet/roslyn/issues/26626
```C#
public class C
{
    string x { get; set; } = null; // expect a warning
}
```

Found in nullable dogfood.

Note: there are some IDE tests (in `CSharpDeclareAsNullableCodeFixTests`) for this bug, which should be unskipped.

**Comment by jcouv** [comment]
Note that `string x =>  null;` correctly reports a warning.

**Comment by sharwell** [comment]
I would not expect a normal null-assignment warning in this case on the use of `null`. The code example is equivalent to the following:

```csharp
public class C
{
  string x { get; set; }
}
```

The warning should be one of the following:

* Simply ignored, and report a warning if the property is not set in all constructors
* Reported as a warning that the value is equal to the default initialization behavior, so the code fix for the warning can be implemented as a removal of the initialization as opposed to marking the property as nullable

**Comment by jcouv** [comment]
The same problem occurs for fields:
`string field = null; // I'd expect a warning`

**Comment by cston** [comment]
It looks like the warnings are reported if there is an explicit constructor, but warnings are not reported when there is only an implicit default constructor.

**Comment by cston** [comment]
See #31139.

**Comment by jaredpar** [comment]
This is fixed now. Need to add a regression test.

**Comment by cston** [comment]
The original bug still repros. There is no warning for the initializer of `x` since warnings are not reported for initializers if there is only the default parameter-less constructor.
